rs/tests/boundary_nodes/integration_test_common/src/lib.rs

```diff
@@ -55,165 +50,295 @@ use serde::Deserialize;
-const CANISTER_RETRY_BACKOFF: Duration = Duration::from_secs(2);
-
 fn runtime() -> Runtime {
     Runtime::new().expect("Could not create tokio runtime")
 }
 
-async fn install_canister(env: TestEnv, logger: Logger, path: &str) -> Result<Principal, Error> {
-    let install_node = env
-        .topology_snapshot()
-        .subnets()
-        .next()
-        .unwrap()
-        .nodes()
-        .next()
-        .map(|node| (node.get_public_url(), node.effective_canister_id()))
-        .unwrap();
-
-    info!(
-        &logger,
-        "creating replica agent {}",
-        install_node.0.as_str()
-    );
-    let agent = assert_create_agent(install_node.0.as_str()).await;
+async fn install_counter_canister(env: TestEnv, logger: Logger) -> Result<Principal, Error> {
+    info!(&logger, "creating management agent");
+    let (install_url, effective_canister_id) =
+        get_install_url(&env).expect("failed to get install url");
 
-    let canister = load_wasm(path);
+    let agent = assert_create_agent(install_url.as_str()).await;
 
-    info!(&logger, "installing canister from path {}", path);
-    let canister_id = create_canister(&agent, install_node.1, &canister, None)
-        .await
-        .expect("Could not create http_counter canister");
+    info!(&logger, "creating canister");
+    let cid = create_canister(
+        &agent,
+        effective_canister_id,
+        wat::parse_str(COUNTER_CANISTER_WAT).unwrap().as_slice(),
+        None,
+    )
+    .await
+    .map_err(|err| anyhow!(format!("failed to create canister: {}", err)))?;
 
     info!(&logger, "Waiting for canisters to finish installing...");
     retry_with_msg_async!(
         format!(
             "agent of {} observes canister module {}",
-            install_node.0.to_string(),
-            canister_id.to_string()
+            install_url.to_string(),
+            cid.to_string()
         ),
         &logger,
         READY_WAIT_TIMEOUT,
         RETRY_BACKOFF,
         || async {
-            match agent_observes_canister_module(&agent, &canister_id).await {
+            match agent_observes_canister_module(&agent, &cid).await {
                 true => Ok(()),
                 false => panic!("Canister module not available yet"),
             }
         }
     )
-    .await
-    .unwrap();
+    .await?;
 
-    info!(&logger, "created canister {canister_id}");
+    info!(&logger, "created canister {cid}");
 
-    Ok::<_, Error>(canister_id)
+    Ok::<_, Error>(cid)
 }
 
 /* tag::catalog[]
-Title:: Boundary nodes binary canister test
+Title:: Boundary Nodes API Endpoints Test
 
-Goal:: Install and query a binary canister
+Goal:: Check all API endpoints (status/read_state/query/call)
 
 Runbook:
-. Set up a subnet with 4 nodes and a boundary node.
-
-Success:: The canister installs successfully and calls against it
-return the expected responses
-
-Coverage:: binary canisters behave as expected
+There are 5 subtests:
```

Comment:
```suggestion
There are 6 subtests:
```

Reply:
IMO it would be more elegant to spawn tests in parallel [via testing framework interface](https://github.com/dfinity/ic/blob/2244bbbcbc0021c896b71f12c84c7454214f15db/rs/tests/driver/src/driver/group.rs#L477). We do it e.g. [here](https://github.com/dfinity/ic/blob/2244bbbcbc0021c896b71f12c84c7454214f15db/rs/tests/boundary_nodes/bn_integration_test.rs#L30). Test functions would be nicely isolated, it would be easier to understand the logic and no need to spawn futures/tasks manually. An additional important benefit  is that test summary would report each test individually, hence it's easier to identify a failing subtest and fix it.

Reply:
I split the newly added `api_endpoints_test` into separate tests. I didn't split the other existing tests and also didn't split the `http_endpoints_test` as it reuses uploaded assets across the different subtests and therefore more work would be required to split it.